aws/s3/s3_concurrent.go

```diff
@@ -165,33 +190,58 @@ func (s *S3Concurrent) GetAllConcurrently(bucket, version string, objects []type
 	return s.manager.Process(processFunc, objects)
 }
 
-// getWorker retrieves a worker from the manager's worker pool.
-func (cm *ConcurrencyManager) getWorker() *worker {
-	return <-cm.workerPool
+// getWorker retrieves a number of workers from the manager's worker pool.
+func (cm *ConcurrencyManager) getWorkers(number int) []*worker {
+	cm.workerPool.mutex.Lock()
+	defer cm.workerPool.mutex.Unlock()
+
+	workers := make([]*worker, number)
+	for i := 0; i < number; i++ {
+		workers[i] = <-cm.workerPool.channel
+	}
+	return workers
 }
 
 // returnWorker returns a worker to the manager's worker pool.
 func (cm *ConcurrencyManager) returnWorker(w *worker) {
-	cm.workerPool <- w
+	cm.workerPool.channel <- w
 }
 
-// secureMemory secures enough memory from the manager's memory pool
-// to fit the specified size.
-func (cm *ConcurrencyManager) secureMemory(size int64) {
+// secureMemory secures the memory needed for the given objects
+// from the manager's memory pool.
+func (cm *ConcurrencyManager) secureMemory(objects []types.Object) {
+	cm.memoryPool.mutex.Lock()
+	defer cm.memoryPool.mutex.Unlock()
 
-	var securedMemory int64 = 0
-	for securedMemory < size {
-		securedMemory += <-cm.memoryPool
+	for _, o := range objects {
+		var securedMemory int64 = 0
+		for securedMemory < o.Size {
```

Comment:
Same as the comment below. (Sorry I commented the code below first)

Reply:
I updated the below bit, but in this case still need a for loop to receive from the channel, so wasn't much point.